exchange/bidder_test.go

```diff
@@ -2221,6 +2221,32 @@ func TestTimeoutNotificationOn(t *testing.T) {
 	assert.EqualValues(t, logExpected, logActual)
 }
 
+func TestTmaxInDoRequest(t *testing.T) {
+	ctx, cancelFunc := context.WithDeadline(context.Background(), time.Now().Add(50*time.Second))
+	defer cancelFunc()
+
+	bidRequest := adapters.RequestData{
+		Method: "POST",
+		Uri:    "test.url.com",
+		Body:   []byte(`{"id":"this-id","app":{"publisher":{"id":"pub-id"}}}`),
+	}
+
+	bidderAdapter := bidderAdapter{
+		me: &metricsConfig.NilMetricsEngine{},
+	}
+	logger := func(msg string, args ...interface{}) {}
+
+	httpCallInfo := bidderAdapter.doRequestImpl(ctx, &bidRequest, logger, time.Now(), &config.TmaxAdjustments{Enabled: true, BidderLatencyAdjustment: 20000, BidderResponseMin: 40000})
+	assert.NotNil(t, httpCallInfo.err)
+	assert.Equal(t, ErrTmaxAdjustment, httpCallInfo.err)
+
+	httpCallInfo = bidderAdapter.doRequestImpl(ctx, &bidRequest, logger, time.Now(), &config.TmaxAdjustments{Enabled: true, BidderLatencyAdjustment: 20000, BidderResponseMin: 10000})
+	assert.NotEqual(t, ErrTmaxAdjustment, httpCallInfo.err)
+
+	httpCallInfo = bidderAdapter.doRequestImpl(ctx, &bidRequest, logger, time.Now(), &config.TmaxAdjustments{Enabled: false, BidderLatencyAdjustment: 20000, BidderResponseMin: 40000})
+	assert.NotEqual(t, ErrTmaxAdjustment, httpCallInfo.err)
```

Comment:
just a suggestion. we could also make use of test tables to reduce line of code

Reply:
hmm. I thought about it. These are just 7 lines for code. Setting test case in test tables would also amount to these many lines so I kinda feel that this is more cleaner to do it this way than setting up the test table. Also assertions for 1st test case is different than other two and that will add the complexity.